### PR TITLE
Nsmith/issue 5438 commonize window shard

### PR DESCRIPTION
**Pull request by nsmithtt**

### CODE REVIEW

File: tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_max_pool.py [hunk 1]
```diff
@@ -322,7 +252,7 @@ def copy_input_to_device(self, input: ttl.tensor.Tensor):
         in_n = self.sliding_window_op_params.batch_size
         in_h = self.sliding_window_op_params.input_h
         in_w = self.sliding_window_op_params.input_w
-        assert in_c % 16 == 0, "Input channels should be multiple of 32. General case is TODO"
+        assert in_c % 32 == 0, "Input channels should be multiple of 32. General case is TODO"
```
Comment (mywoodstock) [inline tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_max_pool.py:255]: Why is this change needed? This will disable the shallow depth maxpool?

File: ttnn/ttnn/operations/conv2d.py [hunk 2]
```diff
@@ -133,5 +133,15 @@ def copy_input_to_device(self, input: ttnn.Tensor):
     def copy_output_from_device(self, output: ttnn.Tensor):
         return ttnn.Tensor(self.conv.copy_output_from_device(output.value))
 
+    @ttnn.register_operation(name="ttnn.Conv2d.get_num_cores_nhw", validate_input_tensors=lambda *args, **kwargs: True)
```
Comment (arakhmati) [inline ttnn/ttnn/operations/conv2d.py:136]: Operations are functions/methods that take in and return tensors. So please remove register_operation from here

**Review by arakhmati — CHANGES_REQUESTED** [review]

File: tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_max_pool.py (hunk 1 above)
Comment (nsmithtt) [reply tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_max_pool.py:255]: Doesn't this mean that metal would have to support half tiles?  I didn't see the below code pad the input, unless this is automatically handled?

File: ttnn/ttnn/operations/conv2d.py (hunk 2 above)
Comment (nsmithtt) [reply ttnn/ttnn/operations/conv2d.py]: Removed, thanks!

File: tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_max_pool.py (hunk 1 above)
Comment (nsmithtt) [reply tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_max_pool.py]: Reverted it back! Still curious about my question though :)

**Review by mywoodstock — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Review by tt-nshanker — APPROVED** [review]
